### PR TITLE
VALIS-4 rearrange tracks

### PR DESCRIPTION
**Pull request by emro**
This update allows you to click the tops and bottoms of tracks to move them up and down in the browser, reordering the browser files.

Changes are adopted largely from George's commit: 
https://github.com/VALIS-software/valis-hpgv/commit/e1bd3bebea1e7a7036757891a997ca6d2ed7dc86

### CODE REVIEW

File: src/ui/TrackViewer.tsx [hunk 1]
```diff
@@ -153,6 +156,7 @@ export class TrackViewer extends Object2D {
         // hide/show add panel button
         let clampToTracks = state.clampToTracks == null ? false : state.clampToTracks;
         this.allowNewPanels = state.allowNewPanels == null ? false : state.allowNewPanels;
+        this.reorderTracks = state.reorderTracks == null ? false : state.reorderTracks;
```
Comment (zoldello) [inline src/ui/TrackViewer.tsx:159]: This is fine for consistency. But I would of had done:

this.reorderTracks = !!state.reorderTracks;

File: src/ui/TrackViewer.tsx [hunk 2]
```diff
@@ -405,6 +413,38 @@ export class TrackViewer extends Object2D {
         this.layoutTrackRows(animate);
     }
 
+    setTrackIndex(track: Track, index: number, animate = true) {
+        // re-orders tracks in this.tracks[] and then calls this.layoutTrackRows(animate);
+
+        // apply array bounds check
+        index = Math.min(Math.max(index, 0), this.tracks.length - 1);
```
Comment (zoldello) [inline src/ui/TrackViewer.tsx:420]: Is it a good idea to mutate an argument?

**Review by zoldello — APPROVED** [review]
I have a question about the ENCODE work but its not a show-stopper.

File: src/ui/TrackViewer.tsx (hunk 1 above)
Comment (emro) [reply src/ui/TrackViewer.tsx]: This is better.

File: src/ui/TrackViewer.tsx (hunk 2 above)
Comment (emro) [reply src/ui/TrackViewer.tsx]: No! Updated.

**Review by zoldello — APPROVED** [review]
Looks good.

**Comment by zoldello** [comment]
I fixed a merge conflict

**Pull request merged**